catalogue_graph/tests/adapters/utils/test_iceberg_config.py

```diff
@@ -105,3 +107,94 @@ def test_get_table_loads_when_flag_false(local_catalog_params):  # type: ignore
 
     # Identifiers match and we can assert the metadata location is identical, indicating load not recreate
     assert created.metadata_location == loaded.metadata_location
+
+
+def test_get_iceberg_table_local(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test get_iceberg_table with local configuration."""
+    namespace = "test_ns"
+    table_name = "test_table"
+    db_name = "test_db"
+
+    config = IcebergTableConfig(
+        table_name=table_name,
+        namespace=namespace,
+        use_rest_api_table=False,
+        db_name=db_name,
+    )
+
+    # Mock get_local_table to verify it's called with correct params
+    def mock_get_local_table(
+        table_name: str,
+        namespace: str,
+        db_name: str,
+        schema: Schema | None = None,
+        partition_spec: PartitionSpec | None = None,
+        create_if_not_exists: bool = True,
+    ) -> str:
+        assert table_name == config.table_name
+        assert namespace == config.namespace
+        assert db_name == config.db_name
+        assert create_if_not_exists == config.create_if_not_exists
+        return "mock_table"
+
+    monkeypatch.setattr("adapters.utils.iceberg.get_local_table", mock_get_local_table)
+
+    result = get_iceberg_table(config)
```

Comment:
Could use `assert_called_once_with` , rather than hand-rolling a mock function.  It would also ensure that it's not being called more times than it should be.
```suggestion
        from unittest import mock
    mock_get_local_table = mock.Mock(return_value="mock_table")

    monkeypatch.setattr("adapters.utils.iceberg.get_local_table", mock_get_local_table)

    result = get_iceberg_table(config)

    mock_get_local_table.assert_called_once_with(
        table_name=config.table_name,
        namespace=config.namespace,
        db_name=config.db_name,
        create_if_not_exists=config.create_if_not_exists,
        schema=mock.ANY,
        partition_spec=mock.ANY
    )
```